src/perf_mainloop.cc

```diff
@@ -205,13 +246,16 @@ DDRes worker_loop(DDProfContext &ctx, const WorkerAttr *attr,
   // Setup poll() to watch perf_event file descriptors
   int const pe_len = ctx.worker_ctx.pevent_hdr.size;
   // one extra slot in pfd to accommodate for signal fd
-  struct pollfd poll_fds[k_max_nb_perf_event_open + 1];
+  pollfd poll_fds[k_max_nb_perf_event_open + 1];
   int pfd_len = 0;
   pollfd_setup(&ctx.worker_ctx.pevent_hdr, poll_fds, &pfd_len);
 
   DDRES_CHECK_FWD(signalfd_setup(&poll_fds[pfd_len]));
   int const signal_pos = pfd_len++;
 
+  WorkerServer const server =
```

Comment:
Somehow I had not thought about making this a separate thread. I thought we would make it live within the mainloop.
I think it is better this way.